CHANGELOG.md

```diff
@@ -33,6 +33,35 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   }
   ```
 
+- Add a fraction-based sizing strategy. ([#168])
+
+  `ConstrainedSizeFraction` describes the layout of children which occupy a fraction of their parents' layout space. 
+
+  Previously, this effect could be achieved by composing a `ConstrainedSize` layout as a child of a `GeometryReader`.
+
+  ```swift
+  GeometryReader { geometry in
+    Row { row in 
+
+      ...
+
+      let availableWidth = geometry.constraint.width.maximum
+      row.addFixed(child: self.element.constrainedTo(width: .absolute(width * 0.25)))
```

Comment:
These examples are not functionally equivalent. `Row` measures its children twice: 
1. constrained by the full width of the row, to get "basis sizes"
2. constrained to the actual allocated width, to get the height

During the first measure, you'll get the same result, but during the second one, the child will wind up being doubly constrained to 1/16 of the available width. If the child is flexible, like a label, it could result in an incorrect height.

Here's an example that demonstrates the difference:
```swift
    var element: Element {
        Column { col in
            col.minimumVerticalSpacing = 1
            col.add(child: a)
            col.add(child: b)
        }
        .centered()
    }

    var a: Element {
        GeometryReader { geometry in
            let width = geometry.constraint.width.maximum
            return Row { row in
                row.add(child: Spacer(width: 300, height: 20).box(background: .red))
                row.add(child: Label(text: "a b c d e f g h i j k l m n o")
                    .box(background: .green)
                    .constrainedTo(width: .absolute(width * 0.25)))
            }
        }
    }

    var b: Element {
        Row { row in
            row.add(child: Spacer(width: 300, height: 20).box(background: .red))
            row.add(child: Label(text: "a b c d e f g h i j k l m n o")
                .box(background: .green)
                .constrainedTo(fractionalWidth: 0.25))
        }
    }
```

Looks like this:

<img width="386" alt="Screen Shot 2020-10-19 at 6 10 14 PM" src="https://user-images.githubusercontent.com/100192/96527982-5cd61500-1236-11eb-9dc6-7f488b77bc17.png">

Reply:
The second sizing pass is going to be a potentially unresolvable problem for us. 

We assume that provided `SizeConstraint` is always the size constraint of the parent, which appears to be too naive. Do you have any ideas as to an alternative?

Reply:
Yeah, the measure function's contract is more like "given this much space, how big are you". There's no guarantee that the constraint is the size of the parent.

The GeometryReader solution seems reasonable in this case, where you actually want to base your size on the constraint passed to the row (not the constraints the row passes to its children).

Alternatively, if the main use case you're trying to solve is making it easier to create proportionally-divided stacks, you might want something new that works more like `EqualStack`. (Or maybe add ratio values to `EqualStack`.)

Reply:
OK, I'm going to close this one for now and consider such a path forward when I next have time to do so. Thanks Andrew.

Reply:
Sure thing, sorry for being the bearer of bad news! Happy to talk through this any time.

Reply:
Oh, please don't be sorry! I'm grateful for your insight, and not at all deterred from potentially landing a contribution that fulfills the semantics I'm looking for.